src/vs/workbench/parts/output/common/outputServices.ts

```diff
@@ -155,44 +145,36 @@ export class OutputService implements IOutputService {
 		return Object.keys(this.receivedOutput);
 	}
 
-	public clearOutput(channel = DEFAULT_OUTPUT_CHANNEL): void {
-		this.receivedOutput[channel] = '';
+	public getActiveChannel(): string {
+		if (!this.outputPanel || !this.outputPanel.getInput()) {
+			return null;
+		}
 
-		this._onOutput.fire({ channel: channel, output: null /* indicator to clear output */ });
+		return (<OutputEditorInput>this.outputPanel.getInput()).getChannel();
 	}
 
-	public showOutput(channel: string = DEFAULT_OUTPUT_CHANNEL, sideBySide?: boolean | Position, preserveFocus?: boolean): TPromise<IEditor> {
-
-		// If already opened, focus it unless we want to preserve focus
-		let existingOutputEditor = this.findOutputEditor(channel);
-		if (existingOutputEditor) {
-			if (!preserveFocus) {
-				return this.editorService.focusEditor(existingOutputEditor);
-			}
-
-			// Still reveal last line
-			(<OutputEditorInput>existingOutputEditor.input).revealLastLine();
+	public clearOutput(channel: string): void {
+		this.receivedOutput[channel] = '';
 
-			return Promise.as(existingOutputEditor);
-		}
+		this._onOutput.fire({ channel: channel, output: null /* indicator to clear output */ });
+	}
 
-		// Otherwise open new
-		return this.editorService.openEditor(OutputEditorInput.getInstance(this.instantiationService, channel), preserveFocus ? EditorOptions.create({ preserveFocus: true }) : null, <any>sideBySide);
+	public showOutput(channel: string, preserveFocus?: boolean): TPromise<IEditor> {
+		return this.panelService.openPanel(OUTPUT_PANEL_ID, !preserveFocus).then((panel: OutputPanel) => {
+			this.outputPanel = panel;
+			return this.outputPanel.setInput(OutputEditorInput.getInstance(this.instantiationService, channel), EditorOptions.create({ preserveFocus: true })).
+				then(() => this.outputPanel);
+		});
 	}
 
-	private findOutputEditor(channel: string): IEditor {
-		let editors = this.editorService.getVisibleEditors();
-		for (let i = 0; i < editors.length; i++) {
-			let editor = editors[i];
-			if (editor.input instanceof OutputEditorInput && (<OutputEditorInput>editor.input).getChannel() === channel && (<OutputEditorInput>editor.input).getMime() === OUTPUT_MIME) {
-				return editor;
-			}
+	public revealLastLine(): void {
+		if (this.outputPanel) {
+			this.outputPanel.revealLastLine();
 		}
-
-		return null;
 	}
 
 	public dispose(): void {
+		this.outputPanel.dispose();
```

Comment:
@isidorn suggest to not dispose outputPanel here, it will be handled when the compositePart is disposed